model/project_ref.go

```diff
@@ -284,7 +284,11 @@ func (p *ProjectRef) AddToRepoScope(user *user.DBUser) error {
 				Enabled: true,
 			}}
 			// creates scope and give user admin access to repo
-			return errors.Wrapf(repoRef.Add(user), "problem adding new repo ref for '%s/%s'", p.Owner, p.Repo)
+			err = repoRef.Add(user)
+			if err != nil {
+				return errors.Wrapf(err, "problem adding new repo ref for '%s/%s'", p.Owner, p.Repo)
+			}
+			return errors.Wrapf(p.Update(), "problem updating project ref '%s' while adding to repo scope", p.Id)
```

Comment:
The reason I initially didn't update here is because we want the whole projectRef update to happen at once. `AddToRepoScope` is used in two places: in [the route|https://github.com/evergreen-ci/evergreen/blob/ebd7a65c5f2af4feb0c19837c9c4b173cd5af4e8/rest/route/project.go#L466] we call UpdateProject immediately after, but for [the UI|https://github.com/evergreen-ci/evergreen/blob/ebd7a65c5f2af4feb0c19837c9c4b173cd5af4e8/service/project.go#L735] we update the repo way after we Upsert the project. Rather than Update directly in here and update the project twice in each case, I would rearrange the UI function.